modules/andes-core/broker/src/main/java/org/wso2/andes/subscription/LocalSubscription.java

```diff
@@ -90,11 +90,16 @@ public LocalSubscription(OutboundSubscription subscription, String subscriptionI
 
         this.subscription = subscription;
 
-        //We need to keep the subscriptionType in basic subscription for notification
-        if(subscription instanceof AMQPLocalSubscription) {
-            setSubscriptionType(SubscriptionType.AMQP);
-        } else if(subscription instanceof MQTTLocalSubscription) {
-            setSubscriptionType(SubscriptionType.MQTT);
+        // In case of mock subscriptions, this becomes null
+        if (null != subscription) {
+            //We need to keep the subscriptionType in basic subscription for notification
+            if(subscription instanceof AMQPLocalSubscription) {
+                setSubscriptionType(SubscriptionType.AMQP);
+            } else if(subscription instanceof MQTTLocalSubscription) {
+                setSubscriptionType(SubscriptionType.MQTT);
```

Comment:
This is used when resolving wild card subscriptions which are different in MQTT and AMQP. Since both MQTT and AMQP subscriptions have to extend BasicSubscription which is not an interface, we need to change the behaviour of the AndesSubscription class hierarchy. Therefore, It is better to invest some time on this.